juniper_official/routing/collect-sr-color-lsp-stats.rule

```diff
@@ -18,41 +19,66 @@
 healthbot {
     topic routing.spring {
         rule collect-sr-color-lsp-stats {
-            keys [ lsp-name ];
+            keys [ lsp-name lsp-cfg-name ];
             description "Collects SR color LSP stats periodically";
             sensor jnpr-sr-color-lsp {
                 open-config {
                     sensor-name /junos/services/segment-routing/traffic-engineering/ingress/usage/;
                     frequency 60s;
                 }
             }
+            sensor cisco-oc-sr-color-lsp {
+                description "Cisco open-config SR Color LSP sensor definition";
+                open-config {
+                    sensor-name /interfaces/interface/state/counters;
+                    frequency 60s;
+                }
+            }
             field lsp-destination {
                 sensor jnpr-sr-color-lsp {
                     where "/mpls/signaling-protocols/segment-routing/sr-te-ip-policies/sr-te-ip-policy/@to-address =~ /{{input-to-address}}/";
                     path "/mpls/signaling-protocols/segment-routing/sr-te-ip-policies/sr-te-ip-policy/@to-address";
-                }                
+                } 
+                sensor cisco-oc-sr-color-lsp {
+                    path cisco_lsp_destination_dummy_path;
+                    data-if-missing {
+                        value "cisco_sr_lsp_destination";
+                    }
+                }               
                 type string;
                 description "LSP destination in case of SR color LSP and name in case of SR LSP";
             }
             field lsp-color {
                 sensor jnpr-sr-color-lsp {
                     where "/mpls/signaling-protocols/segment-routing/sr-te-ip-policies/sr-te-ip-policy/@color =~ /{{input-color}}/";
                     path "/mpls/signaling-protocols/segment-routing/sr-te-ip-policies/sr-te-ip-policy/@color";
-                }                
+                }
+                sensor cisco-oc-sr-color-lsp {
+                    path cisco_lsp_color_dummy_path;
+                    data-if-missing {
```

Comment:
data if missing makes the data to buffer for 1.5 times the sensor frequency. Which means though we know that the data would come of this point we will have to buffer it for 90s which is very inefficient way to process at scale.

Ideal case would be is to write the 2 rules and for different vendors where key attributes like this differ.